### PR TITLE
Update Mastodon API version string to include Mastodon version compatibility

### LINKED ISSUE #11264: Return Pleroma-like version in /api/v1/instance
### Describe the feature you'd like

In `/api/v1/instance`, Pleroma [exposes software version in a format](https://docs.pleroma.social/backend/development/API/differences_in_mastoapi_responses/) consisting of Mastodon version it is compliant with, the name of software and its version, like `2.7.2 (compatible; Pleroma 2.4.51-477-g2744c4c5-develop`. Currently, it's not possible to identify Friendica just using the version field. This format is also used by other Fediverse platforms implementing Mastodon API, like [Mitra](https://codeberg.org/silverpill/mitra/src/branch/main/src/mastodon_api/instance/types.rs#L22). This might be helpful for clients supporting multiple backends.

It might also be helpful to provide a semver-like version software, I think simply using something like 2022.5.0 would be enough

### PR DESCRIPTION
**Pull request by MrPetovan**
Closes #11264

### CODE REVIEW

File: src/Module/Api/Mastodon/Instance.php [hunk 1]
```diff
@@ -22,6 +22,7 @@
 namespace Friendica\Module\Api\Mastodon;
 
 use Friendica\Core\System;
+use Friendica\DI;
```
Comment (nupplaphil) [inline src/Module/Api/Mastodon/Instance.php]: Please use DI directly per constructor :)

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Module/Api/Mastodon/Instance.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Api/Mastodon/Instance.php]: I do not follow you, the `BaseApi` constructor doesn't have `DI` parameter?

Comment (nupplaphil) [reply src/Module/Api/Mastodon/Instance.php]: No, sry what I mean is that you can replace all `DI::*` calls with a (maybe extended) parameter of the `__constructor()` of this method, so you don't need this import at all
--> we're already inside the DI, so no extra DI call necessary :)

Comment (MrPetovan) [reply src/Module/Api/Mastodon/Instance.php]: Ah, I get it now, thanks.

**Comment by MrPetovan** [comment]
I think I got the hang of it, it doesn't horribly crashes at least. It took me a while to figure out where the `$server` constructor parameter was coming from, but I finally tracked it down in `Router::getModule`. Once I figured this out, I knew I could just add the required parameters before `$server` in the module constructor and Bob's your uncle.

**Comment by annando** [comment]
While this PR contains an improvement concerning the classes, I don't really see the need for the change concerning the version string. I haven't seen any system that rejected an API request because of the version number.

**Comment by MrPetovan** [comment]
It isn't about rejection, more like about standardization.

**Review by annando — APPROVED** [review]

**Comment by annando** [comment]
I still don't like it, but okay.

**Review by nupplaphil — APPROVED** [review]

**Comment by MrPetovan** [comment]
Can I merge myself with two approvals?

**Pull request merged**